DuckDuckGo/AppConfigurationFetch.swift

```diff
@@ -313,3 +300,78 @@ class AppConfigurationFetch {
+                                                       refreshStartDate: Date,
+                                                       task: BGTask,
+                                                       status: BackgroundRefreshCompletionStatus,
+                                                       previousStatus: BackgroundRefreshCompletionStatus?) -> BackgroundRefreshCompletionStatus {
+
+        task.setTaskCompleted(success: status.success)
+        scheduleBackgroundRefreshTask()
+
+        let refreshEndDate = Date()
+        let difference = refreshEndDate.timeIntervalSince(refreshStartDate)
+        backgroundFetchTaskDuration += Int(difference)
+
+        guard let last = previousStatus else {
+            return status
+        }
+
+        var mutableStore = store
+
+        // If a new completion status comes in then the counter for the previous status should be reversed.
+        switch last {
+        case .expired:
+            mutableStore.backgroundFetchTaskExpirationCount = max(0, store.backgroundFetchTaskExpirationCount - 1)
+        case .noData:
+            mutableStore.backgroundNoDataCount = max(0, store.backgroundNoDataCount - 1)
+        case .newData:
+            mutableStore.backgroundNewDataCount = max(0, store.backgroundNewDataCount - 1)
+        }
+
+        return status
+    }
+
+    @available(iOS 13.0, *)
+    static func backgroundRefreshTaskHandler(store: AppConfigurationFetchStatistics,
+                                             configurationFetcher: AppConfigurationFetch,
+                                             queue: DispatchQueue,
+                                             task: BGTask) {
+
+        let refreshStartDate = Date()
+        var lastCompletionStatus: BackgroundRefreshCompletionStatus?
+
+        // taskCompletion gets called at the end of the refresh process, either by the task being expired by the OS or by the refresh process
+        // completing successfully. It checks whether it has been called earlier in the same refresh run and self-corrects its statistics if so.
+
+        task.expirationHandler = {
+            var mutableStore = store
+            mutableStore.backgroundFetchTaskExpirationCount += 1
+
+            DispatchQueue.main.async {
+                lastCompletionStatus = backgroundRefreshTaskCompletionHandler(store: store,
+                                                                              refreshStartDate: refreshStartDate,
+                                                                              task: task,
+                                                                              status: .expired,
+                                                                              previousStatus: lastCompletionStatus)
+            }
+        }
+
+        queue.async {
+            let newData = configurationFetcher.fetchConfigurationFiles(isBackground: true)
```

Comment:
Since it's a Bool which is returned, why don't naming it like `let newDataExists = ...` ?
Wdyt?

Reply:
Good suggestion!